src/items/functions.md

```diff
@@ -203,12 +203,34 @@ extern "C" fn new_i32() -> i32 { 0 }
 let fptr: extern "C" fn() -> i32 = new_i32;
 ```
 
-Functions with an ABI that differs from `"Rust"` do not support unwinding in the
-exact same way that Rust does. Therefore, unwinding past the end of functions
-with such ABIs causes the process to abort.
+### Unwinding
+
+Most ABI strings come in two variants, one with an `-unwind` suffix and one without.
+The `Rust` ABI always permits unwinding, so there is no `Rust-unwind` ABI.
+
+The choice of ABI, together with the [panic mode][panic-modes], determines the
+behavior when unwinding out of a function.
+
+In the table below, "Foreign unwind (unforced)" refers to something like a C++
+exception; the table indicates the behavior of an unwinding
+operation reaching each type of ABI boundary (function
+declaration or definition using the corresponding ABI string).
+Additionally, `C` in the ABI strings may be substituted with
+`stdcall` or any other ABI supported by the language
+implementation.
+
+| panic runtime  | ABI          | `panic`-unwind                        | Foreign unwind (unforced) |
+| -------------- | ------------ | ------------------------------------- | ----------------------- |
+| `panic=unwind` | `"C-unwind"` | unwind                                | unwind                  |
+| `panic=unwind` | `"C"`        | abort                                 | [Undefined Behavior]    |
+| `panic=abort`  | `"C-unwind"` | `panic!` aborts (no unwinding occurs) | abort                   |
+| `panic=abort`  | `"C"`        | `panic!` aborts (no unwinding occurs) | [Undefined Behavior]    |
 
 > **Note**: The LLVM backend of the `rustc` implementation
-aborts the process by executing an illegal instruction.
+> aborts the process by executing an illegal instruction.
```

Comment:
I expect this applies to the "abort" entries in the table above, but maybe not to "`panic!` aborts" – can't that result in the `abort` libc function getting called?

Reply:
Oops, thanks for catching the formatting issue.

I am pretty sure that the text is correct as-is, and that in general we do _not_ rely on `libc` to implement `panic=abort`. @Amanieu, @nbdd0121, can one of you confirm?

Reply:
The use of `abort` seems overloaded here.
* An `panic=abort` panic calls `libc::abort()` (done by panic_abort crate).
* Unwinding to a Rust-defined `extern "C"` function causes the process to terminate (`std::terminate` in C++). We currently implement this by calling `panic_cannot_unwind` which performs a nounwind panic. This will cause `libc::abort()` with std (this time done directly in std), but may have different behaviour if a panic handler is used in `no_std` environment (e.g. infinite loop). We used to generate illegal instruction for this but we've changed it a long time ago.

Reply:
Yea, it seems like this should be removed.

I think this was mostly relevant to the 1.24 release where https://github.com/rust-lang/rust/pull/46833 was reverted in 1.24.1, but the docs didn't get reverted.

Reply:
Done!